core/cap-0063.md

```diff
@@ -0,0 +1,447 @@
++    uint32 txMaxInMemoryReadEntries;
++    // Maximum number of disk entry read operations per transaction
++    uint32 txMaxDiskReadEntries;
++    // Maximum number of bytes of disk reads that can be performed per transaction
++    uint32 txMaxDiskReadBytes;
++    // Maximum number of ledger entry write operations per transaction
++    uint32 txMaxWriteLedgerEntries;
++    // Maximum number of bytes that can be written per transaction
++    uint32 txMaxWriteBytes;
++
++    int64 feeDiskReadLedgerEntry;  // Fee per disk ledger entry read
++    int64 feeDiskRead1KB;          // Fee for reading 1KB disk
++    int64 feeWriteLedgerEntry;     // Fee per ledger entry write
++
++    // The following parameters determine the write fee per 1KB.
++    // Write fee grows linearly until bucket list reaches this size
++    int64 bucketListTargetSizeBytes;
++    // Fee per 1KB write when the bucket list is empty
++    int64 writeFee1KBBucketListLow;
++    // Fee per 1KB write when the bucket list has reached `bucketListTargetSizeBytes` 
++    int64 writeFee1KBBucketListHigh;
++    // Write fee multiplier for any additional data past the first `bucketListTargetSizeBytes`
++    uint32 bucketListWriteFeeGrowthFactor;
++};
++
+ // Historical data (pushed to core archives) settings for contracts.
+ struct ConfigSettingContractHistoricalDataV0
+ {
+@@ -302,7 +342,8 @@ enum ConfigSettingID
+     CONFIG_SETTING_STATE_ARCHIVAL = 10,
+     CONFIG_SETTING_CONTRACT_EXECUTION_LANES = 11,
+     CONFIG_SETTING_BUCKETLIST_SIZE_WINDOW = 12,
+-    CONFIG_SETTING_EVICTION_ITERATOR = 13
++    CONFIG_SETTING_EVICTION_ITERATOR = 13,
++    CONFIG_SETTING_CONTRACT_LEDGER_COST_V1 = 14
+ };
+ 
+ union ConfigSettingEntry switch (ConfigSettingID configSettingID)
+@@ -335,5 +376,7 @@ case CONFIG_SETTING_BUCKETLIST_SIZE_WINDOW:
+     uint64 bucketListSizeWindow<>;
+ case CONFIG_SETTING_EVICTION_ITERATOR:
+     EvictionIterator evictionIterator;
++case CONFIG_SETTING_CONTRACT_LEDGER_COST_V1:
++    ConfigSettingContractLedgerCostV1 contractLedgerCost;
+ };
+ }
+diff --git a/Stellar-transaction.x b/Stellar-transaction.x
+index 7d32481..f9bdc1f 100644
+--- a/Stellar-transaction.x
++++ b/Stellar-transaction.x
+@@ -882,8 +882,8 @@ struct SorobanResources
+     // The maximum number of instructions this transaction can use
+     uint32 instructions; 
+ 
+-    // The maximum number of bytes this transaction can read from ledger
+-    uint32 readBytes;
++    // The maximum number of bytes this transaction can read from disk backed entries
++    uint32 diskReadBytes;
+     // The maximum number of bytes this transaction can write to ledger
+     uint32 writeBytes;
+ };
+ 
++struct SorobanResourcesExtV0
++{
++    // Vector of uint16_t indices representing what Soroban
++    // entries in the footprint are evicted, based on the
++    // order of keys provided in the readWrite footprint.
++    opaque<> evictedSorobanEntries;
```

Comment:
Should we call this `archivedEntries` dropping `Soroban` in the name since its already inside _Soroban_ResourcesExtV0 struct?